src/library/librarycontrol.cpp

```diff
@@ -404,6 +415,12 @@ void LibraryControl::emitKeyEvent(QKeyEvent&& event) {
     }
 }
 
+void LibraryControl::refocusLibrary() {
+    if (!m_pLibraryWidget->hasFocus() && !m_pSidebarWidget->hasFocus()) {
```

Comment:
While checking the code it seems that both m_pLibraryWidget and m_pSidebarWidget might be null, we need to check for this.

Could the check of m_pSidebarWidget be moved into setLibraryFocus(), i.e. by replacing `if (m_pSidebarWidget) {` with `(m_pSidebarWidget && !m_pSidebarWidget->hasFocus()) {` there (see below)? Explicitly focusing the sidebar widget in setLibraryFocus() if it already has the focus doesn't make sense to me.

Then this condition could be simplified to `if (m_pLibraryWidget && !m_pLibraryWidget->hasFocus()) {`.

Reply:
Thx, for the hint, ooks like you're right.
However, as far as I investigated, the functions that make use of the original `LibraryControl::setLibraryFocus()` do some existence/focus tests themselves. Maybe some of them can be consolidated.

Reply:
it's `LibraryControl::emitKeyEvent(QKeyEvent&& event)`
and `LibraryControl::slotGoToItem(double v)`

Reply:
I'm also in favor of renaming `setLibraryFocus` to `focusTracksTable`. Apart from track tables no library features' (Playlists, Crates, Computer) top pages can't be focused anyway